### PR TITLE
Refactor/localopt specific imports

### PR DESCRIPTION
**Pull request by jlnav**
Verified locally works with openmpi

### CODE REVIEW

File: libensemble/gen_funcs/aposmm_localopt_support.py [hunk 1]
```diff
@@ -202,6 +195,8 @@ def run_local_nlopt(user_specs, comm_queue, x0, f0, child_can_read, parent_can_r
     Runs an NLopt local optimization run starting at ``x0``, governed by the
     parameters in ``user_specs``.
     """
+    import nlopt
```
Comment (jmlarson1) [inline libensemble/gen_funcs/aposmm_localopt_support.py:198]: Can we ensure that moving this import to so late doesn't cause issues when trying to run with (say) 64 workers?

Comment (jmlarson1) [reply libensemble/gen_funcs/aposmm_localopt_support.py:198]: There was no issue in my tests.

**Comment by shuds13** [comment]
I don't see that the breaking change is necessary. The old rc options would still work - just putting imports in functions and removing the defaults. Either way, we also need a test, and to document when a global import should not be used.

I have a test that uses OpenMPI with the executor, which I can add, so this can be run in the CI. 

Also, mpi4py is working on my laptop, with or without global import, is it failing for you? @jlnav

**Review by shuds13 — CHANGES_REQUESTED** [review]
see comment

**Pull request closed without merging**